http/client.go

```diff
@@ -16,37 +17,89 @@ import (
 // New creates a new http.Client with a custom certificate, which can be loaded from the passed CA Bundle file and/or
 // directory. If both CABundleFile and CABundleDir are empty arguments, it creates an unsecure HTTP client.
 func New(CABundleFile, CABundleDir string, httpTimeout time.Duration) (*http.Client, error) {
+	return _new(CABundleFile, CABundleDir, httpTimeout, "")
+}
+
+// NewAcceptInvalidHostname new http.Client with ability to accept HTTPS certificates that don't
+// match the hostname of the server they are connecting to.
+func NewAcceptInvalidHostname(CABundleFile, CABundleDir string, httpTimeout time.Duration, hostname string) (*http.Client, error) {
+	return _new(CABundleFile, CABundleDir, httpTimeout, hostname)
+}
+
+func _new(CABundleFile, CABundleDir string, httpTimeout time.Duration, acceptInvalidHostname string) (*http.Client, error) {
```

Comment:
just a small comment: maybe **acceptInvalidHostname** would be better named something like "alternateHostname"

Reply:
I rather the prior as it explicitly says it comes from its caller `NewAcceptInvalidHostname` in comparison to its `New` counterpart

Reply:
but if you have a strong opinion be my guest